### PR TITLE
Examples clean up and test fixes

### LINKED ISSUE #1157: test-select-cpu fails on Ubuntu 16.04 for i686
This issue was reported to me by an Ubuntu contributor. Since Ubuntu 16.04 will become the next LTS, it might be worth also testing future releases of AF for it.

The current Debian package fails to build on Ubuntu 16.04 on their 32-bit architecture (i686). The other architectures build fine. Please find below the relevant section of the build log containing the offending test. Sounds like another tolerance related error.

```
36/89 Test #41: Test_select_cpu ..................***Failed    5.09 sec
Running main() from gtest_main.cc
[==========] Running 37 tests from 13 test cases.
[----------] Global test environment set-up.
[----------] 3 tests from Select/0, where TypeParam = float
[ RUN      ] Select/0.Simple
[       OK ] Select/0.Simple (85 ms)
[ RUN      ] Select/0.RightScalar
[       OK ] Select/0.RightScalar (67 ms)
[ RUN      ] Select/0.LeftScalar
[       OK ] Select/0.LeftScalar (65 ms)
[----------] 3 tests from Select/0 (217 ms total)

[----------] 3 tests from Select/1, where TypeParam = double
[ RUN      ] Select/1.Simple
[       OK ] Select/1.Simple (151 ms)
[ RUN      ] Select/1.RightScalar
[       OK ] Select/1.RightScalar (106 ms)
[ RUN      ] Select/1.LeftScalar
[       OK ] Select/1.LeftScalar (101 ms)
[----------] 3 tests from Select/1 (358 ms total)

[----------] 3 tests from Select/2, where TypeParam = af::af_cfloat
[ RUN      ] Select/2.Simple
[       OK ] Select/2.Simple (180 ms)
[ RUN      ] Select/2.RightScalar
[       OK ] Select/2.RightScalar (129 ms)
[ RUN      ] Select/2.LeftScalar
[       OK ] Select/2.LeftScalar (130 ms)
[----------] 3 tests from Select/2 (439 ms total)

[----------] 3 tests from Select/3, where TypeParam = af::af_cdouble
[ RUN      ] Select/3.Simple
[       OK ] Select/3.Simple (342 ms)
[ RUN      ] Select/3.RightScalar
[       OK ] Select/3.RightScalar (241 ms)
[ RUN      ] Select/3.LeftScalar
[       OK ] Select/3.LeftScalar (233 ms)
[----------] 3 tests from Select/3 (816 ms total)

[----------] 3 tests from Select/4, where TypeParam = unsigned int
[ RUN      ] Select/4.Simple
[       OK ] Select/4.Simple (155 ms)
[ RUN      ] Select/4.RightScalar
[       OK ] Select/4.RightScalar (116 ms)
[ RUN      ] Select/4.LeftScalar
[       OK ] Select/4.LeftScalar (116 ms)
[----------] 3 tests from Select/4 (387 ms total)

[----------] 3 tests from Select/5, where TypeParam = int
[ RUN      ] Select/5.Simple
[       OK ] Select/5.Simple (204 ms)
[ RUN      ] Select/5.RightScalar
[       OK ] Select/5.RightScalar (134 ms)
[ RUN      ] Select/5.LeftScalar
[       OK ] Select/5.LeftScalar (134 ms)
[----------] 3 tests from Select/5 (472 ms total)

[----------] 3 tests from Select/6, where TypeParam = long long
[ RUN      ] Select/6.Simple
[       OK ] Select/6.Simple (345 ms)
[ RUN      ] Select/6.RightScalar
/build/arrayfire-3.2.1+dfsg1/test/select.cpp:83: Failure
Value of: hcond[i] ? ha[i] : b
  Actual: 4.76578e+18
Expected: hc[i]
Which is: 4765783585667689730
[  FAILED  ] Select/6.RightScalar, where TypeParam = long long (218 ms)
[ RUN      ] Select/6.LeftScalar
/build/arrayfire-3.2.1+dfsg1/test/select.cpp:83: Failure
Value of: hcond[i] ? ha[i] : b
  Actual: 8.60665e+18
Expected: hc[i]
Which is: 8606648235303615511
[  FAILED  ] Select/6.LeftScalar, where TypeParam = long long (218 ms)
[----------] 3 tests from Select/6 (781 ms total)

[----------] 3 tests from Select/7, where TypeParam = unsigned long long
[ RUN      ] Select/7.Simple
[       OK ] Select/7.Simple (317 ms)
[ RUN      ] Select/7.RightScalar
[       OK ] Select/7.RightScalar (219 ms)
[ RUN      ] Select/7.LeftScalar
[       OK ] Select/7.LeftScalar (219 ms)
[----------] 3 tests from Select/7 (755 ms total)

[----------] 3 tests from Select/8, where TypeParam = unsigned char
[ RUN      ] Select/8.Simple
[       OK ] Select/8.Simple (71 ms)
[ RUN      ] Select/8.RightScalar
[       OK ] Select/8.RightScalar (51 ms)
[ RUN      ] Select/8.LeftScalar
[       OK ] Select/8.LeftScalar (50 ms)
[----------] 3 tests from Select/8 (172 ms total)

[----------] 3 tests from Select/9, where TypeParam = char
[ RUN      ] Select/9.Simple
[       OK ] Select/9.Simple (78 ms)
[ RUN      ] Select/9.RightScalar
[       OK ] Select/9.RightScalar (62 ms)
[ RUN      ] Select/9.LeftScalar
[       OK ] Select/9.LeftScalar (62 ms)
[----------] 3 tests from Select/9 (202 ms total)

[----------] 3 tests from Select/10, where TypeParam = short
[ RUN      ] Select/10.Simple
[       OK ] Select/10.Simple (73 ms)
[ RUN      ] Select/10.RightScalar
[       OK ] Select/10.RightScalar (51 ms)
[ RUN      ] Select/10.LeftScalar
[       OK ] Select/10.LeftScalar (49 ms)
[----------] 3 tests from Select/10 (173 ms total)

[----------] 3 tests from Select/11, where TypeParam = unsigned short
[ RUN      ] Select/11.Simple
[       OK ] Select/11.Simple (71 ms)
[ RUN      ] Select/11.RightScalar
[       OK ] Select/11.RightScalar (50 ms)
[ RUN      ] Select/11.LeftScalar
[       OK ] Select/11.LeftScalar (50 ms)
[----------] 3 tests from Select/11 (171 ms total)

[----------] 1 test from Select
[ RUN      ] Select.NaN
[       OK ] Select.NaN (131 ms)
[----------] 1 test from Select (131 ms total)

[----------] Global test environment tear-down
[==========] 37 tests from 13 test cases ran. (5076 ms total)
[  PASSED  ] 35 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] Select/6.RightScalar, where TypeParam = long long
[  FAILED  ] Select/6.LeftScalar, where TypeParam = long long

 2 FAILED TESTS
```

### PR DESCRIPTION
**Pull request by pavanky**
[skip ci]

- Removes unnecessary console options
- Removes unnecessary `getchar()` for windows
- Fixes #1157

### CODE REVIEW

File: examples/graphics/surface.cpp [hunk 1]
```diff
@@ -13,8 +13,8 @@
 
 using namespace af;
 
-static const int ITERATIONS = 30;
-static const float PRECISION = 1.0f/ITERATIONS;
+static const int POINTS = 30;
```
Comment (shehzan10) [inline examples/graphics/surface.cpp]: POINTS is a typedef to a struct on Windows in windef.h. That's causing build failure.

Comment (pavanky) [inline examples/graphics/surface.cpp]: Ofcourse :|

**Comment by umar456** [comment]
Lets put `af/macros.h` into the examples folder. I don't think it needs to be in include/af

**Comment by shehzan10** [comment]
For the record, we will be adding `/subsystem:console` with the visual studio project files that will be shipped with the ArrayFire installer.

The purpose of the getchar was to prevent the command prompts from exiting without user input with from without the VS debugger. This can be achieved by using `/subsystem:console` in Linker->System->Subsystem. Hence removing the code within the windows macro.

**Pull request merged**